runtime-modules/project-token/src/lib.rs

```diff
@@ -1086,17 +1086,19 @@ impl<T: Config>
         start: Option<T::BlockNumber>,
         duration: T::BlockNumber,
         allocation_source: T::AccountId,
-        allocation_amount: JoyBalanceOf<T>,
+        nominal_allocation_amount: JoyBalanceOf<T>,
     ) -> DispatchResult {
+        let token_info = Self::ensure_token_exists(token_id)?;
+        token_info.revenue_split.ensure_inactive::<T>()?;
+
+        let allocation_amount = token_info
+            .revenue_split_rate
+            .mul_floor(nominal_allocation_amount);
         ensure!(
             !allocation_amount.is_zero(),
             Error::<T>::CannotIssueSplitWithZeroAllocationAmount,
         );
```

Comment:
> I could not understand the first part, I also added a test which (if I understood it properly) should test your first assertion and it work as expected.
As of now the creator retains in his account (100% - revenue_split_rate) * allocation
Can you rephrase maybe?

The amount for the split comes from channel account balance, so if you have, let's say, 10% `split_rate` and you issue a revenue split, 90% will remain in the channel account and the creator won't be able to access it. What we want is to allocate 10% for the split and **send** the rest to the creator (member controller account in case of member channel and probably council budget in case of curator channel), so this is what is missing. It should be added in `content` pallet.

Reply:
See: https://github.com/Joystream/joystream/blob/carthage/runtime-modules/content/src/lib.rs#L2692